esmvalcore/preprocessor/_derive/__init__.py

```diff
@@ -31,55 +34,63 @@ def _get_all_derived_variables():
     return derivers
 
 
-ALL_DERIVED_VARIABLES = _get_all_derived_variables()
+ALL_DERIVED_VARIABLES: dict[str, type[DerivedVariableBase]] = (
+    _get_all_derived_variables()
+)
 
 __all__ = list(ALL_DERIVED_VARIABLES)
 
 
-def get_required(short_name, project):
+def get_required(short_name: FacetValue, project: FacetValue) -> list[Facets]:
     """Return all required variables for derivation.
 
-    Get all information (at least `short_name`) required for derivation.
+    Get all information (at least ``short_name``) required for derivation.
 
     Parameters
     ----------
-    short_name : str
-        `short_name` of the variable to derive.
-    project : str
-        `project` of the variable to derive.
+    short_name:
+        Short name of the variable to derive.
+    project:
+        Project of the variable to derive.
 
     Returns
     -------
-    list
-        List of dictionaries (including at least the key `short_name`).
+    list[esmvalcore.typing.Facets]
+        List of facets (including at least the key ``short_name``).
+
     """
+    short_name = str(short_name)
```

Comment:
Please raise a TypeError (or wait for the subsequently called code to raise a TypeError) instead of silently converting, as this is unlikely to produce the correct result

Reply:
Removed cast in https://github.com/ESMValGroup/ESMValCore/pull/2772/commits/ecbecc6e3fec487238daeae25e3d65c09f0e37fd#diff-06aaaf773b663b9bb8097147f72e0e66759aee5ee00c0f0d6db45144157cf327